src/data/finset/basic.lean

```diff
@@ -1142,6 +1142,14 @@ end
 @[simp] lemma sdiff_subset (s t : finset α) : s \ t ⊆ s :=
 show s \ t ≤ s, from sdiff_le
 
+lemma sdiff_ssubset [decidable_eq α] {x y : finset α} (hx : y ⊆ x) (hy : y.nonempty) :
+  x \ y ⊂ x :=
+begin
+  obtain ⟨i, hi⟩ := hy,
+  rw ssubset_iff_of_subset (sdiff_subset _ _),
+  exact ⟨i, hx hi, λ t, (mem_sdiff.1 t).2 hi⟩,
+end
```

Comment:
Is `sdiff_lt` true on a general boolean_algebra (with `y \ne \bot`)? If so, can you prove it there instead?

Reply:
Indeed it is! A bit harder though.